### PR TITLE
Document resetting Rock-on Root (#434)

### LINKED ISSUE #434: Document resetting rock-ons-root
Thanks to form member Jorma_Tuomainen for highlighting this issue. We have existing fragility that can be worked around by a rock-ons-root share reset/switch-out - assuming our delete-rockon command line script can successfully remove all Rock-ons db assets associated with the prior rock-ons-root (see: https://github.com/rockstor/rockstor-core/issues/2549).

Forum reference: https://forum.rockstor.com/t/unknown-internal-error-doing-a-get-to-api-rockons/8797

### PR DESCRIPTION
**Pull request by Hooverdan96**
Fixes #434 .

### This pull request's proposal (superseeding PR #502).
add the Rock-on root reset process to the section that deals with uninstalling/installing Rock-ons. Also, some minor cleanups in the same file.

### Checklist
- [X] With the proposed changes no Sphinx errors or warnings are generated.
- [X] I have added my name to the AUTHORS file, if required (descending alphabetical order).

**Note:** Linkcheck will continue to fail until a currently open PR has been merged.

### CODE REVIEW

**Comment by phillxnet** [comment]
@Hooverdan96 Re:

> **Note:** Linkcheck will continue to fail until a currently open PR has been merged.

This also requires that this branch be re-based. But not required as we know the failure now. And I have now merged that PR. So we are hopefully getting ahead of the little thorn.

About to review this now so no need to re-base by the way.

File: interface/overview.rst [hunk 1]
```diff
@@ -39,11 +39,12 @@ specifically for this purpose.
...
 
-It is assumed you have already setup your :ref:`pools` and one or more
-shares in those pools (see our :ref:`createshare`) appropriate for your
-Rock-ons, i.e. a plex-movies share and a plex-config share.
+It is assumed that :ref:`pools` and one or more shares in those pools
+(see our :ref:`createshare`) have already been created, appropriate for the
+Rock-on requirements, e.g. a *plex-movies* share and a *plex-config* share.
 
-But we also need to create the :ref:`rockons_root` share.
+As a one-time setup activity, it is also necessary create
```
Comment (phillxnet) [inline interface/overview.rst:46]:
```suggestion
As a one-time setup activity, it is also necessary to create
```

File: interface/overview.rst [hunk 2]
```diff
@@ -53,14 +54,14 @@ All Rock-ons require the **Rock-on service** to be enabled and prior to
...
 configuring a sufficiently large share for the rock-ons to be installed into.
 
-We recommend creating a *dedicated* share to be used as the rock-ons root.
-While any share can be chosen as the rock-ons root, we **strongly** recommend
-to create a share to be used *only* as the rock-ons root to prevent any
-conflict that may arise with a mixed-used share. The rock-ons root share is
+It is recommended to create a *dedicated* share to be used as the Rock-ons root.
+While any share can be chosen as the Rock-ons root, it is **strongly** recommended
+to create a share to be used **only** as the Rock-ons root to prevent any
+conflict that may arise with a mixed-used share. The Rock-ons root share is
```
Comment (phillxnet) [inline interface/overview.rst:60]:
```suggestion
conflict that may arise with a mixed-use share. The Rock-ons root share is
```

@Hooverdan96 I know this was the case before, but just noting while I'm here.

File: interface/overview.rst [hunk 3]
```diff
@@ -53,14 +54,14 @@ All Rock-ons require the **Rock-on service** to be enabled and prior to
...
 used by Docker to store permanent data such as images and container
 layers. As a result, any conflict or alteration of these data that might result
 from a mixed used share has the potential to break the installed Rock-ons.
```
Comment (phillxnet) [inline interface/overview.rst:63]:
```suggestion
from a mixed-use share has the potential to break the installed Rock-ons.
```

Again this does not pertain to your alterations but we could slip this in as we go.

File: interface/overview.rst [hunk 4]
```diff
@@ -496,6 +497,67 @@ For the Rock-on named *Emby server*, for instance, the commands would be:
 
 *This example is for newer Rockstor versions.*
 
+.. _rockons_root_reset:
+
+Reset Rock-on Root
```
Comment (phillxnet) [inline interface/overview.rst:502]:
```suggestion
Reset The Rock-ons Root
```

@Hooverdan96 I prefer this reference as it is the root for all rock-ons, not for a single rock-on. And our pics and other doc references hopefully mostly conform to this arbitrary but important reference. [EDIT] ~~I know a 'the' could be nice but we are titles here so I think this floats :) .~~

File: interface/overview.rst [hunk 5]
```diff
@@ -496,6 +497,67 @@ For the Rock-on named *Emby server*, for instance, the commands would be:
...
+    *recreate* Rock-ons root option described below.
+
+* Uninstall all installed Rock-ons (likely requires the command line as described 
+  in :ref:`rockons_force_delete`).
+
+.. note::
+
+   All shares that were used for any of the Rock-ons will not be affected by these process
+   steps, so a later re-installation of an associated Rock-on should put the situation
+   right back to where it was before the Rock-on root reset process.
```
Comment (phillxnet) [inline interface/overview.rst:538]:
```suggestion
   right back to where it was before the Rock-ons root reset process.
```

File: interface/overview.rst [hunk 6]
```diff
@@ -496,6 +497,67 @@ For the Rock-on named *Emby server*, for instance, the commands would be:
...
+
+* Turn off Rock-ons Service (either on the Rockons page directly, or via the
```
Comment (phillxnet) [inline interface/overview.rst:540]:
```suggestion
* Turn off the Rock-ons Service (either on the Rockons page directly, or via the 
```

File: interface/overview.rst [hunk 7]
```diff
@@ -496,6 +497,67 @@ For the Rock-on named *Emby server*, for instance, the commands would be:
...
+
+.. note::
+
+   All shares that were used for any of the Rock-ons will not be affected by these process
+   steps, so a later re-installation of an associated Rock-on should put the situation
+   right back to where it was before the Rock-on root reset process.
+
+* Turn off Rock-ons Service (either on the Rockons page directly, or via the 
+  **System --> Services** web page).
+* Following :ref:`rockons_root`, either create a new Rock-on root share with
```
Comment (phillxnet) [inline interface/overview.rst:542]:
```suggestion
* Following :ref:`rockons_root`, to either create a new Rock-ons root share with
```

File: interface/overview.rst [hunk 8]
```diff
@@ -496,6 +497,67 @@ For the Rock-on named *Emby server*, for instance, the commands would be:
...
+  a different name, or delete and recreate the Rock-on root share with its original name.
```
Comment (phillxnet) [inline interface/overview.rst:543]:
```suggestion
  a different name, or delete and recreate the Rock-ons root share with its original name.
```

File: interface/overview.rst [hunk 9]
```diff
@@ -496,6 +497,67 @@ For the Rock-on named *Emby server*, for instance, the commands would be:
...
+
+.. note::
+
+  In case of creating a new Rock-on root share with a different name, adjust the Rock-ons
```
Comment (phillxnet) [inline interface/overview.rst:547]:
```suggestion
  In the case of creating a new Rock-ons root share with a different name, adjust the Rock-ons
```

File: interface/overview.rst [hunk 10]
```diff
@@ -496,6 +497,67 @@ For the Rock-on named *Emby server*, for instance, the commands would be:
...
+  Service setting by selecting the new share name from the dropdown. If the Rock-on root
```
Comment (phillxnet) [inline interface/overview.rst:548]:
```suggestion
  Service setting by selecting the new share name from the dropdown. If the Rock-ons root
```

File: interface/overview.rst [hunk 11]
```diff
@@ -496,6 +497,67 @@ For the Rock-on named *Emby server*, for instance, the commands would be:
...
+  share has been re-created with the same name, no change is necessary. Delete the old
+  Rock-on root share to free up disk space.
+
+* Restart the Rock-ons Service.
+* Run an update on the list of available Rock-ons.
+
+Now the Rock-ons root should be consistent, albeit empty, once again, and (re-)installation
+of Rock-ons should work as before. Or, if the configuration backup was created
+this could now be executed, which should skip all existing configuration elements
+and only result in the re-installation of the previously capture Rock-ons.
```
Comment (phillxnet) [inline interface/overview.rst:558]:
```suggestion
and only result in the re-installation of the previously installed Rock-ons.
```

**Review by phillxnet — CHANGES_REQUESTED** [review]
@Hooverdan96 Thanks for yet-another great addition.

I'm a bit concerned about folks jumping to this brute-force solution: but there are warnings in there. But this is rather the nuclear option in my opinion. But as you indicated here, it can be necessary on occasions. So it can only be good to have it documented as you've presented. Thanks.

See what you think of my suggested changes, almost all trivial actually; but hopefully also nice-to-have.

My main indicator that this this procedure is necessary has been log entries to the effect that a subvol/snapshot was was missing pertaining to a docker container. What we likely need in rockstor-core here is more use of:
```
docker system prune -a --volumes
```
And maybe that could be added to this doc as a pre-nuclear (wipe / recreate) option. And only if that, once run, followed by a Pool refresh to update subvols fails to work do we then suggest this rather radical option of switching/refreshing the Rock-ons root share.

**Comment by Hooverdan96** [comment]
Thanks for the suggestions, I've accepted them. I will incorporate your additional comments from above and bring these back (and address the current Sphinx failure that now seems to have popped up).

**Comment by phillxnet** [comment]
@Hooverdan96 Thanks. Let me know when I should re-test and publish:
Re the Sphinx error:

```
Reset The Rock-ons Root
------------------ [docutils]
/home/runner/work/rockstor-doc/rockstor-doc/interface/overview.rst:503: WARNING: Title underline too short.
```
Yes, that one was down to me I'm afraid. I failed to also modify the number of chars indicating a sub-header to match the sub-header changes I made. My apologies for the extra work there.

**Comment by Hooverdan96** [comment]
@phillxnet, I did end up rebasing the branch, and rearranged the nuclear option to be after a section on pruning, and also updated the logs symptom with your comment about missing subvols.

**Comment by Hooverdan96** [comment]
The linkcheck failures this time around where related to `smartmontools` links, which timed out (first time for me). i assume those are temporary at this point:
```
(    howtos/smart: line    6) timeout   https://www.smartmontools.org/ - HTTPSConnectionPool(host='www.smartmontools.org', port=443): Read timed out. (read timeout=20)
(    howtos/smart: line  191) timeout   https://www.smartmontools.org/browser/trunk/smartmontools/smartctl.8.in - HTTPSConnectionPool(host='www.smartmontools.org', port=443): Read timed out. (read timeout=20)
(    howtos/smart: line   94) timeout   https://www.smartmontools.org/browser/trunk/smartmontools/smartd.conf.5.in - HTTPSConnectionPool(host='www.smartmontools.org', port=443): Read timed out. (read timeout=20)
(    howtos/smart: line  193) timeout   https://www.smartmontools.org/wiki/Supported_RAID-Controllers - HTTPSConnectionPool(host='www.smartmontools.org', port=443): Read timed out. (read timeout=20)
```

**Review by phillxnet — APPROVED** [review]

**Comment by phillxnet** [comment]
@Hooverdan96 Thanks for including that extra command, may end-up saving some folks from having to do the more extensive reset procedure.

I'll merge and get this published. As always much appreciated. It's great to have this procedure finally documented. In time we can automate better a Rock-on reset, or have Web-UI intergration for force uninstall and the like; but bit-by-bit.

**Comment by phillxnet** [comment]
Re the smartmontools linkcheck hick-up, a couple of builds locally did not show this failure. So as you surmised I think we caught them on a bad day. Or they don't like GitHub actions accessing their site.

**Comment by phillxnet** [comment]
PR product PRODUCTION published.

**Pull request merged**